### PR TITLE
Update Delete Command to delete with Person's IC

### LINKED ISSUE #7: As an expert user I can easily delete unnecessary data to reduce clutter in the app
... so that I can maintain a clean and organised patient database.

### PR DESCRIPTION
**Pull request by jiahui0309**
Added support for usage of delete command (Closes https://github.com/AY2324S2-CS2103T-F14-2/tp/issues/7). This PR covers:

- Ensuring that the parsing of arguments with the command word `delete` is updated with the new class.
- Updating Unit tests to ensure that DeleteCommand runs and throws errors where necessary.
- The program can compile and run without a problem.

If one of these points above still has an issue please let me know. The DeleteCommand Class now looks like this:
Screenshot 2024-03-16 at 6 30 33 PM

Note:
- The command has yet to be tried on the actual interface as the AddCommand has to be updated first.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/DeleteCommand.java [hunk 1]
```diff
@@ -4,43 +4,43 @@
...
         requireNonNull(model);
         List<Person> lastShownList = model.getFilteredPersonList();
 
-        if (targetIndex.getZeroBased() >= lastShownList.size()) {
-            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
-        }
+        Person personToDelete = lastShownList.stream()
+                .filter(predicate::test)
+                .findFirst()
+                .orElseThrow(() -> new CommandException(Messages.MESSAGE_NO_MATCHING_IC));
```
Comment (drustanyjt) [inline src/main/java/seedu/address/logic/commands/DeleteCommand.java:42]: @jiahui0309 does this mean that you'll only be able to delete patients that are currently visible on the screen?

**Review by drustanyjt — COMMENTED** [review]
Got a question about lastShownList

File: src/main/java/seedu/address/logic/commands/DeleteCommand.java (hunk 1 above)
Comment (jiahui0309) [reply src/main/java/seedu/address/logic/commands/DeleteCommand.java:42]: Currently, it uses a list of profiles which are previously filtered, but might be better to use the entire database of patients in the list

**Review by drustanyjt — APPROVED** [review]
LGTM!

File: src/main/java/seedu/address/logic/commands/DeleteCommand.java (hunk 1 above)
Comment (drustanyjt) [reply src/main/java/seedu/address/logic/commands/DeleteCommand.java:42]: Agreed, thanks!

**Pull request merged**